File: testdata/jennies/builders/map_index_assignment/JavaBuilder/sandbox/SomeStructBuilder.java
package sandbox;

import java.util.HashMap;

public class SomeStructBuilder implements cog.Builder<SomeStruct> {
    protected final SomeStruct internal;
    
    public SomeStructBuilder() {
        this.internal = new SomeStruct();
    }
    public SomeStructBuilder annotations(String key,String value) {
		if (this.internal.annotations == null) {
			this.internal.annotations = new HashMap<>();
		}
        this.internal.annotations.put(key, value);
        return this;
    }
    public SomeStruct build() {
        return this.internal;
    }
}

File: testdata/jennies/builders/known_any/CSharpBuilder/src/Grafana/Foundation/KnownAny/SomeStructBuilder.cs
namespace Grafana.Foundation.KnownAny;


public class SomeStructBuilder : Cog.IBuilder<SomeStruct>
{
    protected readonly SomeStruct @internal;

    public SomeStructBuilder()
    {
        this.@internal = new SomeStruct();
    }

    public SomeStructBuilder Title(string title)
    {
        if (this.@internal.Config == null)
        {
            this.@internal.Config = new Config();
        }
        ((Config) this.@internal.Config).Title = title;
        return this;
    }

    public SomeStruct Build()
    {
        return this.@internal;
    }
}
